pkg/controller/machine_safety.go

```diff
@@ -93,23 +93,27 @@ func (c *controller) checkAndFreezeORUnfreezeMachineSets(wg *sync.WaitGroup) {
 		// Unfreeze machineset when replica count reaches higherThreshold - SafetyDown
 		lowerThreshold := higherThreshold - c.safetyOptions.SafetyDown
 
-		machineDeployment := c.getMachineDeploymentsForMachineSet(machineSet)[0]
-		if machineDeployment != nil {
+		machineDeployments := c.getMachineDeploymentsForMachineSet(machineSet)
+		if machineDeployments != nil && len(machineDeployments) >= 1 {
+			machineDeployment := machineDeployments[0]
+			if machineDeployment != nil {
```

Comment:
Does this if condition add any value? or am i missing something?

Reply:
This fixes the bug that safety controller panics if a machineset does not have a corresponding machinedeployment.

This is the issue Padma found in safety controller.

Reply:
The panic is "index out of bounds" because machineDeployments array is empty if there are no machinedeployments for the machineset.

Reply:
You are already testing for array length to be greater than or equal to 1 right? so array would atleast have one element.

Reply:
And i understood the purpose of the change, I was concerned about the additional if condition alone.

Reply:
Yeah even I was wondering about the last point. Let's keep it then perhaps, no harm.

Reply:
I have removed the spurious nil check before the len(). But the nested nil check is required, I think.